python/GangaDirac/Lib/Files/DiracFile.py

```diff
@@ -174,8 +174,7 @@ def _attribute_filter__set__(self, name, value):
         if value != "" and value is not None:
             #   Do some checking of the filenames in a subprocess
             if name == 'lfn':
-                if self.namePattern == '':
-                    self.namePattern = os.path.basename(value)
+                self.namePattern = os.path.basename(value)
```

Comment:
yes but you have just requested that the DiracFile have the explicit LFN so you're changing the object implicitly I would argue.
The subfiles of the wildcard all work as expected but if you have ```df=DiracFile(namePattern='foo');``` ```df.lfn=bar``` you would naively assume that ```df.namePattern``` would then be ```basename('bar')```?

I think there isn't a strict matching between LFN and namePattern, but this just makes life difficult if you're uploading a local file to an LFN with a different name.